systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaRebalanceUtils.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.utils.kafkaUtils;
+
+import io.strimzi.systemtest.Constants;
+import io.strimzi.systemtest.resources.crd.KafkaRebalanceResource;
+import io.strimzi.test.TestUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import static io.strimzi.test.k8s.KubeClusterResource.kubeClient;
+
+public class KafkaRebalanceUtils {
+
+    private static final Logger LOGGER = LogManager.getLogger(KafkaRebalanceUtils.class);
+
+    private KafkaRebalanceUtils() {}
+
+    public enum KafkaRebalanceState {
+        PendingProposal,
+        ProposalReady,
+        Rebalancing,
+        Ready,
+        NotReady,
+        Stopped
+    }
+
+    public static void waitForKafkaRebalanceCustomResourceState(String resourceName, KafkaRebalanceState state) {
```

Comment:
Why you didn't use `ResourceManager.waitForResourceStatus()` ?

Reply:
Because KafkaRebalance resource did not match attributes with other resources such as Kafka, KafkaConnect etc.

Reply:
it's something we are going to revert back even because the OpenShift UI is not able to show the right status. @tomncooper are you still working on that?

Reply:
Yes I am working on switching the state and type fields so that they are correct for the OCP UI.